### DISCUSSION TITLE
Saving a route (to bookmarks)

### DISCUSSION THREAD

**Discussion by yasio90** [organicmaps/organicmaps#781] [Ideas]
It would be great if there would be an option of saving a route, possibly to bookmarks. Sometimes I plan a route in the app but don't want to follow it immediately. If I close the app the route gets lost and I need to start from scratch. Probably the easiest way would be to save it as a track in bookmarks - while walking I very rarely follow navigation, I mostly use maps.me as a map, so having a track saved in the bookmarks based on a previously planned route would be for me optimal.

**Comment by ashamanin** [comment]
It would also be nice to have the option to save the last (active) route. Even if it is not bookmarked. For example, the user might exit navigation mode or close the app for some reason. After that, you need to set waypoints again.

**Reply by biodranik** [reply]
How would you implement it for users? Where users can find the last route and select it?

**Reply by ashamanin** [reply]
For example:
- 1st approach: long tap on the map, navi-mod launched, an additional on-screen button appeared (a curved arrow in a circle) . It should load a last route.
- 2-nd approach: option in setting "Last route menu point", if "true" -> show additional menu item "Last route"

**Comment by kcpr** [comment]
I presented similar idea there: https://github.com/organicmaps/organicmaps/discussions/2741 . I also wrote about "recent tracks list", which would possibly (at least partially) be easier to implement I assume. That's because I believe bookmarks are KML-compatible now, so using them for saving a track would possibly require to first convert a track into KML-compatible format and later on to generate a track as described here: #2384 . "Recent track list" may use some internal way of saving a track though.
What do You think about such list? Would it resolve the issue in Your opinion?
I do agree that the end goal would be to be able to save a track as a bookmark, but I just think it may be harder to implement so maybe "recent tracks list" could be something in the middle which would help.
Thank You for sharing this idea by the way.

**Reply by yasio90** [reply]
For my purposes it would be completely sufficient. In most cases I plan a route a day or a few hours before starting following it, so there is no necessity to store it for long. I think it's a very good idea.
